src/searching/peak_finder.rs

```diff
@@ -0,0 +1,58 @@
+// 1-D Array Peak Finder
+pub fn peak_finder_1d_greedy(arr: &[usize]) -> Option<usize> {
+    for (i, value) in arr.iter().enumerate() {
+        if arr.len() == 1
+            || (i == 0 && *value >= arr[i + 1])
+            || (i > 0 && i < (arr.len() - 1) && (arr[i - 1] <= *value && *value >= arr[i + 1])
+                || (i == arr.len() - 1 && *value >= arr[i - 1]))
+        {
+            return Some(*value);
+        } else {
+            continue;
+        }
+    }
+    None
+}
+
+pub fn peak_finder_1d_divide_conquer(arr: &[usize]) -> Option<usize> {
+    let mut _arr = vec![0; arr.len()];
+    _arr.clone_from_slice(arr);
```

Comment:
I think this will take O(n^2) both memory and performance complexity. which will result in very bad performance in large arrays.